embedded/src/main/java/org/wildfly/core/embedded/EmbeddedHostControllerFactory.java

```diff
@@ -98,11 +96,11 @@ private EmbeddedHostControllerFactory() {
     }
 
     public static HostController create(final File jbossHomeDir, final ModuleLoader moduleLoader, final Properties systemProps, final Map<String, String> systemEnv, final String[] cmdargs) {
-        Assert.checkNotNullParam("jbossHomeDir", jbossHomeDir);
-        Assert.checkNotNullParam("moduleLoader", moduleLoader);
-        Assert.checkNotNullParam("systemProps", systemProps);
-        Assert.checkNotNullParam("systemEnv", systemEnv);
-        Assert.checkNotNullParam("cmdargs", cmdargs);
+        assert jbossHomeDir != null : "jbossHomeDir is null";
+        assert moduleLoader != null : "moduleLoader is null";
+        assert systemProps != null : "systemProps is null";
+        assert systemEnv != null : "systemEnv is null";
+        assert cmdargs != null : "cmdargs is null";
```

Comment:
These methods are public but generally shouldn't be used by consumers of the API. Though if they are these assertions will only work if the user enables assertions. Since this, and the `EmbeddedStandaloneFactory`, are only used internally by the `ManagedProcessFactory` this might not matter.